OpenRA.Mods.RA/Activities/Infiltrate.cs

```diff
@@ -50,11 +48,19 @@ protected override void OnInside(Actor self)
 
 			var exp = self.Owner.PlayerActor.TraitOrDefault<PlayerExperience>();
 			if (exp != null)
-				exp.GiveExperience(experience);
+				exp.GiveExperience(trait.Info.PlayerExperience);
 
-			if (!string.IsNullOrEmpty(notification))
+			if (!string.IsNullOrEmpty(trait.Info.Notification))
 				Game.Sound.PlayNotification(self.World.Map.Rules, self.Owner, "Speech",
-					notification, self.Owner.Faction.InternalName);
+					trait.Info.Notification, self.Owner.Faction.InternalName);
+		}
+
+		public override Activity Tick(Actor self)
+		{
+			if (trait.IsTraitDisabled)
+				return CanceledTick(self);
```

Comment:
As I mentioned earlier you need to cancel the activity (`Cancel(self)`).  Subverting the inner activity handling will cause movement glitches or worse.